Some corrections for Mycelium Android

Giszmo opened the pull request:
The above matrix does **not** apply to Mycelium iPhone. That is a completely different app with different features.

Mycelium Android [does support some +90 shitcoins](https://github.com/mycelium-com/wallet-android/blob/76b47a9bb8ba77bf3e07264e0396bb73f4cab5e0/mbw/src/main/java/com/mycelium/wallet/WalletConfiguration.kt#L317) most of which are ERC20. I now see that "ERC20" is not allowed by your schema but I see ERC20 tokens with 5-long symbols like ABYSS, so maybe the schema needs changing, too.

The organization wasn't founded the date presented here with the name it has now. It changed its name and location a few times.

Giszmo committed:
Some corrections for Mycelium Android

The above matrix does **not** apply to Mycelium iPhone. That is a completely different app with different features.

Mycelium Android does support some dozens of shitcoins I don't have handy right now. There is an array in the source code though ...

The organization wasn't founded the date presented here with the name it has now. It changed its name and location a few times.

dcentrica commented on matrix.json:
```diff
@@ -33,6 +33,14 @@
+                    {
+                        "type": "crypto",
+                        "ticker": "ERC20"
+                    },
```
As you've noticed, this is not how the schema works. The Matrix needs to support users selecting a _specific_ coin name from the "currencies" drop down menu at: https://walletmatrix.app/matrix. Yes this can become large and unwieldy, but then vendors need not include _all_ their coins: The addition of currencies into the schema is merely incidental anyway, seeing as WM is really all about Bitcoin.

I have made a note to look at deprecating this for schema v4.

dcentrica replied:
Can you just remove that block with a view to the "currencies" block being deprecated going forward anyway? (This has caused confusion for several vendors previously also)

Giszmo replied:
On the other hand, many people ask for those shitcoins and ERC20 should be a valid option as some wallets allow the user to add their own ERC20 tokens, which makes a lot of sense.

Being a Maxi myself I would be fine with dropping this but it's political and if the goal is to develop a standard by which all wallet providers can describe their feature sets, it should not be pruned. There are products out there that very explicitly do not touch shitcoins and I would like to filter for those. `feature: btcOnly`? Or maybe `coinsSupported < 2`? A numerical `coinsSupported` would make it easier for those shit-coin casinos to get listed without having to keep their selection of the day updated.

Edit: Please also link more prominently to the schema repository. I couldn't find it right now. I'd like to propose some changes/additions.

Giszmo replied:
I removed the ERC20 and added other details

dcentrica replied:
@Giszmo Until yesterday, the goal of these schema was to support consumption into WM. Period. Once more examples of `matrix.json` exist in vendor's public repos, then we can have a conversation as to a). the best method by which these files might be consumed (individually, or through a WM/WS managed API, file or service and b). what a v4 schema might look like. 

ATM the schema is a standard format intended purely for consumption by WM. Eventual consumers (via API for example) needn't be constrained by what is essentially an _internal_ format. Instead and by consensus, a v1,2 (N) API can be designed and built that better suits how third-parties might want to consume it.

I'm super keen to hear further ideas for improvement to the schema - but I'm not seeing right now how the schema _itself_ is going to be useful for anything other than WM.

Question: Are you an official contributer/maintiner to MyCelium for Android? Or do you know them? `matrix.json` should exist in the _vendor_ repo. These decentrica-repos are only a temporary measure until more vendors include it in their own repos.

Giszmo replied:
I worked for Mycelium for 5 years but quit in June to work on WS. They would probably not maintain the matrix file in its current state. With integration in WS and some polished and more popular WM ... maybe.

dcentrica commented on matrix.json:
```diff
@@ -97,28 +102,28 @@
-                "psbt": "?",
-                "reproducible_builds": "?",
+                "psbt": "N",
+                "reproducible_builds": "Y",
```
Excellent! FYI for Android devices, WM parses the JSON file produced by WS for reproducibility. The file is fetched and cached to the local F/S daily.

dcentrica commented on matrix.json:
```diff
@@ -130,12 +135,12 @@
-                "coin_control": "?",
-                "tx_labelling": "?"
+                "coin_control": "N",
+                "tx_labelling": "Y"
```
I use Mycelium and had never even noticed this feature! TIL.

Giszmo replied:
I could tell you about a bunch of other features ... Shamir's Shared Secret? Bit-ID? Address labeling? Change-address type based on receive address type by default? Choose exchange rates from a dozen different sources? Trezor, Ledger and Keepkey support?

Giszmo committed:
refine details

Removed ERC20 as it is not a supported currency. Added others and some fiats.
Mycelium can detect received RBF but not create RBF itself. It can CPFP fee-bump though ...

dcentrica commented on matrix.json:
```diff
@@ -31,19 +31,35 @@
                     {
                         "type": "crypto",
-                        "ticker": "BCH"
+                        "ticker": "BTCV"
```
I will merge this but note that new currencies that don't exist by default in the list of 186 in the existing currencies table (BTCV and countless others) will not automatically make it into user-searches. 

There is a small amount of development work I have been holding off on, to make this happen - namely: A centralised resource of all coins to parse. Seeing as not all exchanges list all coins, this is going to be a little hard to come by I've seen this which can be fetched an cached daily (>8Mb!). Do you know of any other resources? Alts are not my speciality.

Giszmo replied:
I hate alts and prefer to call them shitcoins. I wouldn't lock this field to any list and certainly not an 8MB list. Just let the wallets claim whatever symbols and users search for whatever.

Also if this matrix file is supposed to catch on, give the wallet providers more liberties than what your site can handle. I would probably list some shit-coin-count on WalletScrutiny but certainly not the logos of all those coins or full names etc.
